src/home/getters/get-github-user.ts

```diff
@@ -30,13 +28,22 @@ async function getNewSessionToken(): Promise<string | null> {
   const params = new URLSearchParams(hash.substr(1)); // remove the '#' and parse
   const providerToken = params.get("provider_token");
   if (!providerToken) {
-    return null;
+    const error = params.get("error_description");
+    // supabase auth provider has failed for some reason
+    console.error(`GitHub login provider: ${error}`);
   }
-  return providerToken;
+  return providerToken || null;
 }
 
-async function getNewGitHubUser(providerToken: string): Promise<GitHubUser> {
+async function getNewGitHubUser(providerToken: string | null): Promise<GitHubUser | null> {
   const octokit = new Octokit({ auth: providerToken });
-  const response = (await octokit.request("GET /user")) as GitHubUserResponse;
-  return response.data;
+  try {
+    const response = (await octokit.request("GET /user")) as GitHubUserResponse;
```

Comment:
Is this the best method to use to get the user? Doesn't octokit have a dedicated method?

Reply:
It does but afaik it only works with username, not the user id
https://octokit.github.io/rest.js/v20#users-get-by-username
Using request seems to be the only way